crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -321,13 +321,68 @@ impl KeyValueStore for RocksDb {
                     .into_boxed()
             }
             (Some(prefix), None) => {
-                // start iterating in a certain direction within the keyspace
-                let iter_mode =
-                    IteratorMode::From(prefix, convert_to_rocksdb_direction(direction));
-                let mut opts = ReadOptions::default();
-                opts.set_prefix_same_as_start(true);
+                // RocksDB prefix iteration doesn't support reverse order,
+                // but seeking the start key and iterating in reverse order works.
+                // So we can create a workaround. We need to find the next available
+                // element and use it as an anchor for reverse iteration,
+                // but skip the first element to jump on the previous prefix.
+                // If we can't find the next element, we are at the end of the list,
+                // so we can use `IteratorMode::End` to start reverse iteration.
+                if direction == IterDirection::Reverse {
```

Comment:
I feel like this section could be a little easier to read.

Something like:
```rs
match direction {
    IterDirection::Reverse => self.reverse_prefix_iter(prefix, ...),
    _ => self.prefix_iter(prefix, ...),
}
```

Then inside the new `reverse_prefix_iter`, instead of returning early, just do something like:
```rs
let maybe_next_item = next_prefix(prefix.to_vec())
    .and_then(|next_prefix| self
                            .iter_all(
                                column,
                                Some(next_prefix.as_slice()),
                                None,
                                IterDirection::Forward,
                            )
                            .next())
    .and_then(|res| res.ok());
if let Some(next_item) = maybe_next_item {
    ...
} else {
    ...
}
```
Of course, breaking those down into into smaller functions wouldn't hurt either, but those are the changes that would improve readability the most.